src/convertToObject.js

```diff
@@ -6,7 +6,18 @@
  * @return {object}
  */
 function convertToObject(sourceString) {
-  // write your code here
+  const transformedString = sourceString
+    .split(';')
+    .map((el) => el.split(':').map((el1) => el1.trim()))
+    .filter((el) => el.length > 1);
+
+  const obj = {};
+
+  for (const key in transformedString) {
+    obj[transformedString[key][0]] = transformedString[key][1];
+  }
```

Comment:
[CODE PRACTICE] - Avoid using loops in this task, use iteration methods instead. You can use `reduce` to transform the array into the desired object format. Consider refactoring the for-loop into a `reduce` function to adhere to best practices.